server/test/endpoint_roles.test.js

```diff
@@ -0,0 +1,33 @@
+const { describe, it } = require('mocha')
+const fetch = require('node-fetch')
+const assert = require('assert')
+const { getAdmin1IdPToken } = require('./lib/tokens')
+const { getServerUrl } = require('./lib/urls')
+const dbAdminRole = require('database/layer/admin-client')
+
+describe('Admin Roles API Tests', () => {
+  const url = `${getServerUrl()}`
+  let requestHeaders
+  before(async function () {
+    // GET IDENTITY-PROVIDER TOKEN
+    const userAccessToken = await getAdmin1IdPToken()
+    requestHeaders = {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${userAccessToken}`
+    }
+  })
+
+  describe('Create Roles', () => {
+    it('Should create a new role')
+    it('Should return a 400 for a duplicate role within a client')
```

Comment:
Are we planning to implement these tests under this PR? 
```suggestion
    it('Should create a new role')
    it('Should return a 409 for a duplicate role within a client')
```

Reply:
@k1ch  I will implement these few tests. I have to leave the error code as 400 until we do a further refactor the error handling for these db functions to make use of your new knex patterns.